src/Rules/Arrays/InvalidKeyInArrayDimFetchRule.php

```diff
@@ -34,7 +32,7 @@ public function processNode(Node $node, Scope $scope): array
 		}
 
 		$varType = $scope->getType($node->var);
-		if (count(TypeUtils::getAnyArrays($varType)) === 0) {
+		if (!$varType->isArray()->yes()) {
```

Comment:
Can we have a test that shows whether we used the right choice between `!$varType->isArray()->yes()` / `$varType->isArray()->no()` ? It's not obvious to me which one should be used here.

Reply:
well, turns out that no() was needed to keep the same behaviour as before..